app/models/user.rb

```diff
@@ -9,4 +9,8 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :requests_sent, class_name: 'Friendship', foreign_key: 'user_id'
+  has_many :requests_received, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :friend_requests_sent, through: :requests_sent, source: :request_sender
+  has_many :friend_requests_received, through: :requests_received, source: :request_receiver
```

Comment:
### This is a very good initiative but it's redundant

- [ ] Your user model should have `has_many :friendships` since this state that a user can have many friendships
- [ ] Your user model should also have an `inverted_friendship` bind through a `foreign_key` that should be named `friend_id`
> check out [this link](https://smartfunnycool.com/friendships-in-activerecord/) for more info
> if you need some extra help, don't hesitate to contact me through slack